src/layer/vector/canvas/Path.Canvas.js

```diff
@@ -144,6 +145,18 @@ L.Path = (L.Path.SVG && !window.L_PREFER_CANVAS) || !L.Browser.canvas ? L.Path :
 		}
 	},
 
+	_onDblClick: function (e) {
+		if (this._containsPoint(e.layerPoint)) {
+			this.fire('dblclick', e);
+		}
+	},
+
+	_onContextMenu: function (e) {
+		if (this._containsPoint(e.layerPoint)) {
+			this.fire('contextmenu', e);

```

Comment:
we can get rid of duplication here by using `e.type` instead of a specific event, and using one handler for all three events (can be space-separated with one `on` call)